pkg/framework/context/context.go

```diff
@@ -110,19 +116,33 @@ func (p *Provider) InboundTransportEndpoint() string {
 	return p.inboundTransportEndpoint
 }
 
+func (p *Provider) tryToHandle(svc service.InboundHandler, msg service.DIDCommMsgMap, myDID, theirDID string) error {
+	var opts []interface{}
+	if svcOpts, ok := svc.(service.MessengerOptions); ok {
+		opts = svcOpts.MessengerOptions()
```

Comment:
I only see MessengerOptions being used by `HandleInbound`. It's usage is not so clear since I don't see the HandleInbound code.

Reply:
ok, I moved these options to the `HandleInbound` let me know if it`s ok.